### PR TITLE
[Feat](Nereids) support string functions fold constant by fe 

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
support string functions fold constant by fe:
Concat, Substring, Length, Lower, Upper, Trim, Ltrim, Rtrim, Replace, Left, Right, Locate, 
Instr, Ascii, Bin, ConcatWs, CharacterLength, Initcap, Md5, Md5Sum, 
Field, FindInSet, Repeat, Reverse, Space, SplitByChar, SplitByString, SplitPart, 
SubstringIndex, Strcmp, StrLeft, StrRight, Overlay, ParseUrl, UrlDecode, 
AppendTrailingCharIfAbsent, EndsWith, ExtractUrlParameter, Quote, ReplaceEmpty, Password

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run feut

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run cloud_p0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExecFunction.java [hunk 1]
```diff
@@ -43,4 +43,9 @@
      * return type
      */
     String returnType();
+
+    /**
+     * hasVarArgsc
+     */
+    boolean varArgs();
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExecFunction.java:50]:
```java
boolean varArgs() default false;
```

then you can skip to set varArgs to other functions

Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExecFunction.java]: good idea, done

**Comment by 924060929** [comment]
I think you should check the result type too

**Review by englefly — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
> I think you should check the result type too

we can not get return type before execution 
FunctionSignature signature = new FunctionSignature(fnName, args, null, false);
the third parameter is return expression

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run external

**Review by englefly — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**